src/gui/menu_list.cpp

```diff
@@ -0,0 +1,24 @@
+#include "gui/menu_item.hpp"
+#include "gui/menu_list.hpp"
+#include "gui/menu_manager.hpp"
+
+ListMenu::ListMenu(std::vector<std::string> items_, int* selected_) : 
+  selected(selected_)
+{
+  for(int i = 0; i < items_.size(); i++) {
+    add_entry(i, items_[i]);
+  }
+  add_hl();
+  add_back("OK");
+}
+
+void
+ListMenu::menu_action(MenuItem& item) {
+  // prevent segmentation fault should selected be null
+  if(selected) {
+    *selected = item.get_id();
+  }
```

Comment:
No need to document pointer checks, they're pretty basic and already easy to understand just by looking at the code :)